### PR TITLE
waydroid: 1.3.3 -> 1.3.4

### PR DESCRIPTION
**Pull request by ChemicalXandco**
###### Description of changes

https://github.com/waydroid/waydroid/releases/tag/1.3.4

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by CajuM — CHANGES_REQUESTED** [review]
WayDroid requires sh in its path right now so that should be added.

```patch
diff --git a/pkgs/os-specific/linux/waydroid/default.nix b/pkgs/os-specific/linux/waydroid/default.nix
index ee1230f2e18..d9e1ff3ce7c 100644
--- a/pkgs/os-specific/linux/waydroid/default.nix
+++ b/pkgs/os-specific/linux/waydroid/default.nix
@@ -2,6 +2,7 @@
 , lib
 , fetchFromGitHub
 , python3Packages
+, bash
 , dnsmasq
 , gawk
 , getent
@@ -49,6 +50,7 @@ python3Packages.buildPythonApplication rec {
       python3Packages.gbinder-python
       python3Packages.pygobject3
       python3Packages.pyclip
+      bash
       gawk
       kmod
       lxc
```

It also initializes its data directory on the first run, so the service's start condition is no longer needed. An extra flag `-w` is required when starting the container.

```patch
diff --git a/nixos/modules/virtualisation/waydroid.nix b/nixos/modules/virtualisation/waydroid.nix
index a2cfd806f32..46e5f901015 100644
--- a/nixos/modules/virtualisation/waydroid.nix
+++ b/nixos/modules/virtualisation/waydroid.nix
@@ -56,12 +56,8 @@ in
 
       wantedBy = [ "multi-user.target" ];
 
-      unitConfig = {
-        ConditionPathExists = "/var/lib/waydroid/lxc/waydroid";
-      };
-
       serviceConfig = {
-        ExecStart = "${pkgs.waydroid}/bin/waydroid container start";
+        ExecStart = "${pkgs.waydroid}/bin/waydroid -w container start";
         ExecStop = "${pkgs.waydroid}/bin/waydroid container stop";
         ExecStopPost = "${pkgs.waydroid}/bin/waydroid session stop";
       };
```

**Review by CajuM — APPROVED** [review]
LGTM

File: pkgs/os-specific/linux/waydroid/default.nix [hunk 1]
```diff
@@ -2,6 +2,7 @@
 , lib
 , fetchFromGitHub
 , python3Packages
+, bash
```
Comment (linsui) [inline pkgs/os-specific/linux/waydroid/default.nix:5]: Shouldn't this also be removed?

**Comment by CajuM** [comment]
@SuperSandro2000 sh not being found by waydroid-container.service is still an issue even if all NixOS systems have a `/bin/sh`. As it happens the service doesn't have in its PATH neither `/run/current-system/sw/bin` nor `/bin`. This issue is visible only when initializing Waydroid. `sudo systemctl stop waydroid-container.service && sudo rm -rf /var/lib/waydroid ~/.local/share/waydroid && sudo systemctl start waydroid-container.service && waydroid first-launch` should trigger the bug.

File: pkgs/os-specific/linux/waydroid/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/os-specific/linux/waydroid/default.nix:5]: already did a PR to remove that

**Comment by SuperSandro2000** [comment]
okay, fine, we can add it back with a short comment.

**Pull request merged**